pipeline/src/lib.rs

```diff
@@ -9,13 +9,75 @@ pub mod test_util;
 pub mod util;
 pub mod verify;
 
+use std::collections::BTreeMap;
+use std::sync::{Arc, Mutex};
+
+use serde::de::DeserializeOwned;
+
 pub use pipeline::Pipeline;
 
 pub use powdr_backend::{BackendType, Proof};
 use powdr_executor::witgen::QueryCallback;
 
 use powdr_number::FieldElement;
 
+#[derive(Clone)]
+pub struct HostContext {
+    pub data: Arc<Mutex<BTreeMap<u32, Vec<u8>>>>,
```

Comment:
will the `Input` syscall also be modified to access this instead?
also, maybe rename `data` => `file_descriptors` or `file_data`?

Reply:
also, maybe add a small comment describing the purpose of this struct

Reply:
> will the Input syscall also be modified to access this instead?

Yea that's a good question, I'm not sure yet.